File: Java/DesignPatterns/src/test/java/designPatterns/creational/singleton/MainSingletonTest.java
package designPatterns.creational.singleton;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.*;

class MainSingletonTest {
  private Singleton s1;
  private Singleton s2;

  @BeforeEach
  void setUp() {
    s1 = Singleton.getInstance();
    s2 = Singleton.getInstance();
  }

  @Test
  void equalsTest() {
    assertThat(s1).isEqualTo(s2);
    assertThat(s1.getDate()).isEqualTo(s2.getDate());
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/creational/singleton/MainSingletonTest.kt
package creational.singleton

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MainSingletonTest {

  private lateinit var s1: Singleton
  private lateinit var s2: Singleton

  @BeforeEach
  internal fun setUp() {
    s1 = Singleton.getInstance()!!
    s2 = Singleton.getInstance()!!
  }

  @Test
  internal fun equalsTest() {
    assertThat(s1).isEqualTo(s2)
    assertThat(s1.date).isEqualTo(s2.date)
  }
}